strax/dtypes.py

```diff
@@ -175,14 +175,14 @@ def hitlet_with_data_dtype(n_samples=2):
     return dtype + additional_fields
 
 
-def peak_dtype(n_channels=100, n_sum_wv_samples=200, n_widths=11):
+def peak_dtype(n_channels=100, n_sum_wv_samples=200, n_widths=11, digitize_top=True):
```

Comment:
Shouldnt `digitize_top` default to False so that we preserve previous behavior by default and maintain backward compatibility? or is back compatibility impossible?

Reply:
I was wondering the same and I think that our lineages will not let us do that easily (do I got it right, @WenzDaniel ?).

Reply:
Backward compatibility is not assured as you need the `data_top` field in the array. Otherwise numba doesn't compile.

We could get around this but it would require doubling the numba code or use numba generated jit to get it working. 

Hence, in https://github.com/XENONnT/straxen/pull/1099 I actually worked around this by just dumping the `data_top` field just prior to returning the data.

Reply:
So should we just remove the option if it doesnt work when its set to false?

Reply:
No, because in https://github.com/XENONnT/straxen/pull/1099, I do this option to remove the `data_top`-field to save storage and RAM?

Reply:
okay, let me break it down.

So the current implementation from Dominic doesn't support not having this field in the numba functions he changed here. So, in order for the defaults to work, you need this field. 

Now, imagine that you DON'T want to store top/bottom wf because you run out of storage / it's too intense on your memory / whatever / there is only a top array (in the case of our HE channels). You can do that in two ways: 
1. either change the numba code to also compile on a dataformat that does not have this field. I considered first doing this but I figured that it would result in so much code duplication that I decided against it. Another option was to start returning the result for the top array separately and leave it to the user to add it to the field in peaklets. This was an option but mixing functions that write half of the results in place and the other by returning was in my oppinion less clear.
2. Just add the field `data_top` in the peaklets that enter this part of the numba code. If you desire not to store it, just remove it from the dtype prior to saving it to disk. If this was the default - it's not particularly elegant as you are dragging an extra field along that you don't have to and you also require a second copy step. However, this is not the default and only used for our `_he`-data.

Let's have a chat otherwise if that is useful. I wrote it now down explicitly for bookkeeping

Reply:
Ok, if its easier this way then i am fine with that.